### PR TITLE
#340 Generate rmg saves

### PR DESCRIPTION
**Pull request by langonginc**
#343

### CODE REVIEW

File: src/util/to-rmg.ts [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+
+// add edge
+const addEdge = (graph: MultiDirectedGraph<NodeAttributes, EdgeAttributes, GraphAttributes>, lineId: string) => {
+    const u = graph.extremities(lineId)[0];
+    const v = graph.extremities(lineId)[1];
+    const now = graph.getEdgeAttributes(lineId);
+    const nowStyle: string = now.style;
+    if (u == v) return; // skip u-u
+    if (isColorLine(nowStyle)) {
+        const nowColor = getColor(now);
```
Comment (thekingofcity) [inline src/util/to-rmg.ts:220]: What will happen if `nowColor` is `[]` after `getColor(now)`?

Comment (langonginc) [reply src/util/to-rmg.ts:220]: It won't get a `[]` from `getColor()` if we give a legal line attributes `now`.

I think you may need to  add the following to make sure it is not `[]`

```typescript
if (nowColor == []) return;
```

Comment (thekingofcity) [reply src/util/to-rmg.ts:220]: So as long as the graph is valid, it will never be [] right?

Comment (langonginc) [reply src/util/to-rmg.ts:220]: Yep.

**Review by thekingofcity — APPROVED** [review]
LGTM

**Pull request merged**